File: server-config-examples/lucene/entity-defined-index/src/main/java/examples/springdata/geode/server/lucene/domain/DataCreators.java
package examples.springdata.geode.server.lucene.domain;

import com.github.javafaker.Faker;
import com.github.javafaker.Internet;
import com.github.javafaker.Name;
import examples.springdata.geode.domain.EmailAddress;
import org.springframework.data.repository.CrudRepository;

public class DataCreators {
    public static void createLuceneCustomers(int numberOfCustomer, CrudRepository<Customer, Long> repository) {
        Faker faker = new Faker();
        Name fakerName = faker.name();
        Internet fakerInternet = faker.internet();
        for (int i = 0; i < numberOfCustomer; i++) {
            repository.save(new Customer(i, new EmailAddress(fakerInternet.emailAddress()), fakerName.firstName(), fakerName.lastName()));
        }
    }
}

File: server-config-examples/lucene/entity-defined-index/src/main/kotlin/examples/springdata/geode/server/kt/lucene/domain/DataCreators.kt
package examples.springdata.geode.server.kt.lucene.domain

import com.github.javafaker.Faker
import examples.springdata.geode.domain.EmailAddress
import examples.springdata.geode.server.lucene.domain.JavaLuceneCustomer
import org.springframework.data.repository.CrudRepository

fun createLuceneCustomers(numberOfCustomer: Int, repository: CrudRepository<JavaLuceneCustomer, Long>) {
    val faker = Faker()
    val fakerName = faker.name()
    val fakerInternet = faker.internet()
    (0..numberOfCustomer).forEachIndexed { index, _ ->
        repository.save(JavaLuceneCustomer(index.toLong(),
                EmailAddress(fakerInternet.emailAddress()), fakerName.firstName(), fakerName.lastName()))
    }
}